src/hotspot/share/utilities/filenameUtil.hpp

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#ifndef SHARE_UTILITIES_FILENAMEUTIL_HPP
+#define SHARE_UTILITIES_FILENAMEUTIL_HPP
+
+class FilenameUtil : AllStatic {
+private:
+  static const char* const PidFilenamePlaceholder;
+  static const char* const TimestampFilenamePlaceholder;
+  static const char* const TimestampFormat;
+  static const char* const HostnameFilenamePlaceholder;
+
+  static const size_t StartTimeBufferSize = 20;
+  static const size_t PidBufferSize = 21;
+  static const size_t HostnameBufferSize = 512;
+public:
+  // Expand wildcards in filename:
+  // %p -> PID
+  // %t -> timestamp in YY-MM-DD_HH_MM_SS format
```

Comment:
Not just a "timestamp" though it is specifically the VM start time.

Reply:
The "timestamp" comes from `os::javaTimeMillis()`, it is not the VM start time. 

Unified logging captures the VM start time and uses it from its filename.

```
   output = new LogFileOutput(name, _vm_start_time);
```

Reply:
Sorry then I have misunderstood the refactoring of this. I was expecting "%t" to always mean the VM start time - otherwise what time is it?

Reply:
The time the file was created? The time the file name was "constructed"? Or ?

Reply:
The API does not dictate what timestamp value to use, caller can pass in the timestamp it wants to use. When no value is provided, it uses "current timestamp".

In followup CRs, I don't intend to change the current timestamp sources if there are any.